aries_vcx_core/src/wallet/askar/askar_wallet.rs

```diff
@@ -0,0 +1,285 @@
+
+    pub async fn open(
+        db_url: &str,
+        key_method: Option<StoreKeyMethod>,
+        pass_key: PassKey<'_>,
+        profile: Option<String>,
+    ) -> Result<Self, AriesVcxCoreError> {
+        let backend = Store::open(db_url, key_method, pass_key, profile.clone()).await?;
+
+        Ok(Self { backend, profile })
+    }
+
+    pub async fn create_key(
+        &self,
+        name: &str,
+        alg: KeyAlg,
+        seed: &str,
+        rng_method: RngMethod,
+        metadata: Option<&str>,
+        tags: Option<&[EntryTag]>,
+    ) -> Result<(), AriesVcxCoreError> {
+        let mut session = self.backend.session(self.profile.clone()).await?;
+
+        let key = LocalKey::from_seed(alg, seed.as_bytes(), rng_method.into())?;
+        Ok(session.insert_key(name, &key, metadata, tags, None).await?)
+    }
+
+    pub async fn create_did(&self, key_name: &str) -> Result<(), AriesVcxCoreError> {
+        let mut session = self.backend.session(self.profile.clone()).await?;
+
+        let key_entry = self.fetch_key_entry(&mut session, key_name).await?;
+
+        let local_key = key_entry.load_local_key()?;
+
+        let did_bytes = &local_key.to_public_bytes()?[0..16];
+
+        let did = bs58::encode(did_bytes).into_string();
+        Ok(session
+            .insert("aries-did", &did, &did_bytes, None, None)
+            .await?)
+    }
+
+    async fn fetch_key_entry(
+        &self,
+        session: &mut Session,
+        key_name: &str,
+    ) -> Result<KeyEntry, AriesVcxCoreError> {
+        session.fetch_key(key_name, false).await?.ok_or_else(|| {
+            AriesVcxCoreError::from_msg(
+                AriesVcxCoreErrorKind::WalletRecordNotFound,
+                format!("no key with name '{}' found in wallet", key_name),
+            )
+        })
+    }
+}
+
+#[async_trait]
+impl BaseWallet for AskarWallet {
+    #[cfg(feature = "vdrtools_wallet")]
+    fn get_wallet_handle(&self) -> WalletHandle {
+        todo!("Not yet implemented");
+    }
+
+    // ----- DIDs
+    async fn create_and_store_my_did(
+        &self,
+        seed: Option<&str>,
+        kdf_method_name: Option<&str>,
+    ) -> VcxCoreResult<(String, String)> {
+        todo!("Not yet implemented");
+    }
+
+    async fn key_for_local_did(&self, did: &str) -> VcxCoreResult<String> {
+        todo!("Not yet implemented");
+    }
+
+    // returns new temp_verkey and remembers it internally
+    async fn replace_did_keys_start(&self, target_did: &str) -> VcxCoreResult<String> {
+        todo!("Not yet implemented");
+    }
+
+    // replaces the `target_did`'s current verkey with the one last generated by
+    // `replace_did_keys_start`
+    async fn replace_did_keys_apply(&self, target_did: &str) -> VcxCoreResult<()> {
+        todo!("Not yet implemented");
+    }
+
+    // ---- records
+
+    async fn add_wallet_record(
+        &self,
+        xtype: &str,
+        id: &str,
+        value: &str,
+        tags: Option<HashMap<String, String>>,
```

Comment:
For completeness sake just remind of what we mentioned on the community call earlier.

 The BaseWallet method signatures were catered for the particular vdrtools based wallet implementation. We didn't try to over-generalize for the future at the time we designed `BaseWallet`.
So probably you'll find it desirable to redesign `BaseWallet` trait - you can draw inspiration from this POC https://github.com/hyperledger/aries-vcx/pull/1042  which didn't meant to add support for Askar in particular, but I guess even more ambitiously tried to enable arbitrary wallet implementation in the future. That might or might not been too long shot.